synapse/handlers/federation.py

```diff
@@ -618,6 +618,10 @@ async def _get_events_from_store_or_dest(
         will be omitted from the result. Likewise, any events which turn out not to
         be in the given room.
 
+        This function *does not* recursively get missing auth events of the
+        newly fetched events. Callers must include in the `event_ids` argument
+        any missing events from the auth chain.
```

Comment:
not just missing events

```suggestion
        This function *does not* automatically get missing auth events of the
        newly fetched events. Callers must include the full auth chain of
        of the missing events in the `event_ids` argument, to ensure that any
        missing auth events are correctly fetched.
```